VestingSchedule.sol

```diff
@@ -0,0 +1,227 @@
+    bool private _revocable;
+
+    // revoking address and destination
+    address private _refundDestination;
+    address private _revoker;
+
+    // indicates how much of the vested amount has been released for withdrawal
+    uint256 private _released;
+
+    // indicates if the vesting has been revoked. false by default
+    bool _revoked;
+
+    // the time at which the revocation has taken place
+    uint256 _revokeTime;
+
+    /**
+     * @notice A constructor for initialising a new instance of a Vesting Schedule contract
+     * @param beneficiary address of the beneficiary to whom vested tokens are transferred
+     * @param vestingAmount the amount that is to be vested by the contract
+     * @param vestingCliff duration in seconds of the cliff in which tokens will begin to vest
+     * @param vestingStartTime the time (as Unix time) at which point vesting starts
+     * @param vestingPeriodSec duration in seconds of the period in which the tokens will vest
+     * @param revocable whether the vesting is revocable or not
+     * @param revoker address of the person revoking the vesting
+     * @param refundDestination address of the refund receiver after the vesting is deemed revoked
+     * @param vestingContractOwner the owner that of the vesting contract
+     */
+    constructor (address beneficiary,
+                uint256 vestingAmount,
+                uint256 vestingCliff,
+                uint256 vestingStartTime,
+                uint256 vestingPeriodSec,
+                bool    revocable,
+                address revoker,
+                address refundDestination,               
+                address vestingContractOwner) public {
+
+        // do some basic checks
+        require(vestingAmount > 0, "Amount must be positive");
+        require(beneficiary != address(0), "Beneficiary is the zero address");
+        require(revoker != address(0), "Revoker is the zero address");
+        require(refundDestination != address(0), "Refund destination is the zero address");
+        // solhint-disable-next-line max-line-length
+        require(vestingCliff <= vestingPeriodSec, "Vesting Cliff is longer than duration");
+        // solhint-disable-next-line max-line-length
+        require(vestingPeriodSec > 0, "Vesting duration is 0");
+        // solhint-disable-next-line max-line-length
+         require(vestingStartTime.add(vestingCliff) > block.timestamp, "Final time is before current time");
+
+        _transferOwnership(vestingContractOwner);
+
+        _beneficiary = beneficiary;
+        _vestingAmount = vestingAmount;
+        _revocable = revocable;
+        _duration = vestingPeriodSec;
+        _cliff = vestingStartTime.add(vestingCliff);
+        _start = vestingStartTime;
+        _refundDestination = refundDestination;
+        _revoker = revoker;
+    }
+
+    /**
+     * @return the beneficiary of the tokens.
+     */
+    function beneficiary() external view returns (address) {
```

Comment:
Making the fields public will generate these getter automatically. Is there any reason to not use the default autogenerated ones?

Reply:
done